code/game/machinery/hypnochair.dm

```diff
@@ -0,0 +1,176 @@
+/obj/machinery/hypnochair
+	name = "enhanced interrogation chamber"
+	desc = "A device used to perform \"enhanced interrogation\" through invasive mental conditioning."
+	icon = 'icons/obj/machines/implantchair.dmi'
+	icon_state = "hypnochair"
+	circuit = /obj/item/circuitboard/machine/hypnochair
+	density = TRUE
+	opacity = 0
+	ui_x = 375
+	ui_y = 280
+	var/interrogating = FALSE ///Is the device currently interrogating someone?
+	var/start_time = 0 ///Time when the interrogation was started, to calculate effect in case of interruption
+	var/trigger_phrase = "" ///Trigger phrase to implant
+	var/timerid = 0 ///Timer ID for interrogations
+
+	var/message_cooldown = 0 ///Cooldown for breakout message
+
+/obj/machinery/hypnochair/Initialize()
+	. = ..()
+	open_machine()
+	update_icon()
+
+/obj/machinery/hypnochair/attackby(obj/item/I, mob/user, params)
+	if(!occupant && default_deconstruction_screwdriver(user, icon_state, icon_state, I))
+		update_icon()
+		return
+
+	if(default_pry_open(I))
+		return
+
+	if(default_deconstruction_crowbar(I))
+		return
+
+	return ..()
+
+/obj/machinery/hypnochair/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.notcontained_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "hypnochair", name, ui_x, ui_y, master_ui, state)
+		ui.open()
+
+/obj/machinery/hypnochair/ui_data()
+	var/list/data = list()
+	data["occupied"] = occupant ? 1 : 0
+	data["open"] = state_open
+	data["interrogating"] = interrogating
+
+	data["occupant"] = list()
+	if(occupant)
+		var/mob/living/mob_occupant = occupant
+		data["occupant"]["name"] = mob_occupant.name
+		data["occupant"]["stat"] = mob_occupant.stat
+
+	data["trigger"] = trigger_phrase
+
+	return data
+
+/obj/machinery/hypnochair/ui_act(action, params)
+	if(..())
+		return
+	switch(action)
+		if("door")
+			if(state_open)
+				close_machine()
+			else
+				if(!interrogating)
+					open_machine()
+			. = TRUE
+		if("set_phrase")
+			set_phrase(params["phrase"])
+			. = TRUE
+		if("interrogate")
+			if(!interrogating)
+				interrogate()
+			else
+				interrupt_interrogation()
+			. = TRUE
+
+/obj/machinery/hypnochair/proc/set_phrase(phrase)
+	trigger_phrase = phrase
+
+/obj/machinery/hypnochair/proc/interrogate()
+	if(!trigger_phrase)
+		playsound(get_turf(src), 'sound/machines/buzz-sigh.ogg', 25, TRUE)
+		return
+	var/mob/living/carbon/C = occupant
+	if(!istype(C))
+		playsound(get_turf(src), 'sound/machines/buzz-sigh.ogg', 25, TRUE)
+		return
+	to_chat(C, "<span class='warning'>Strobing coloured lights assault you relentlessly! You're losing your ability to think straight!</span>")
+	interrogating = TRUE
+	start_time = world.time
+	update_icon()
+	timerid = addtimer(CALLBACK(src, .proc/finish_interrogation), 450, TIMER_STOPPABLE)
```

Comment:
what happens if you teleport out of the machine partway through the process?

Reply:
finish_interrogation() checks for occupant, so as long as the base machine code properly unsets occupant on teleportation it'll be fine